backend/src/match/match.service.ts

```diff
@@ -41,18 +53,37 @@ export class MatchService {
 
     await this.removeFromQueue(namespaces, matchedUserId);
 
+    this.logger.info(`${userId} and ${matchedUserId} matched`);
     const matchResult = [
       { userId, socketId },
       { userId: matchedUserId, socketId: matchedUserSocketId },
     ];
-    return { result: matchResult } as Match;
+    return {
+      roomId: nanoid(),
+      result: matchResult,
+      userNames: await this.getUserNames([userId, matchedUserId]),
+    } as Match;
+  }
+
+  async getUserNames(userIds: number[]): Promise<string[]> {
+    const userNames: string[] = [];
+
+    userIds.forEach(async (userId) => {
+      const user = await this.userService.getById(userId);
+      if (user) {
+        userNames.push(user.name);
+      }
+    });
```

Comment:
Let's avoid the `forEach` syntax in general since we can't `await` it and leads to a bug here since the `userNames` array is returned before the names are added.